owncloudApp/src/main/java/com/owncloud/android/presentation/providers/sharing/UsersAndGroupsSearchProvider.kt

```diff
@@ -146,96 +143,96 @@ class UsersAndGroupsSearchProvider : ContentProvider() {
             userQuery,
             REQUESTED_PAGE,
             RESULTS_PER_PAGE
-        )
+        ).also { uiResult ->
+            if (!uiResult.isSuccess() && !uiResult.isLoading()) {
+                showErrorMessage(uiResult.errorMessage)
+            }
 
-        if (!shareesResource.isSuccess()) {
-            showErrorMessage(shareesResource)
-        }
+            val names = uiResult.data
+
+            // convert the responses from the OC server to the expected format
+            if (names?.size!! > 0) {
+                response = MatrixCursor(COLUMNS)
+                val namesIt = names.iterator()
+                var item: JSONObject
+                var displayName: String? = null
+                var icon = 0
+                var dataUri: Uri? = null
+                var count = 0
+
+                val userBaseUri = Uri.Builder().scheme(CONTENT).authority(
+                    suggestAuthority!! + DATA_USER_SUFFIX
+                ).build()
+                val groupBaseUri = Uri.Builder().scheme(CONTENT).authority(
+                    suggestAuthority!! + DATA_GROUP_SUFFIX
+                ).build()
+                val remoteBaseUri = Uri.Builder().scheme(CONTENT).authority(
+                    suggestAuthority!! + DATA_REMOTE_SUFFIX
+                ).build()
+
+                val manager = FileDataStorageManager(
+                    context,
+                    account, context!!.contentResolver
+                )
+                val federatedShareAllowed = manager.getCapability(account!!.name).filesSharingFederationOutgoing
+                    .isTrue
+
+                try {
+                    while (namesIt.hasNext()) {
```

Comment:
I think this file has been translated by Android Studio.

Why don't you use something like a `for (name in names)`